cmp/form/FormModel.ts

```diff
@@ -274,4 +308,83 @@ export class FormModel extends HoistModel {
             }
         );
     }
+
+    private initPersist({
+        fieldsToInclude = null,
+        fieldsToExclude = null,
+        path = 'formValues',
+        ...rootPersistWith
+    }: FormPersistOptions) {
+        const fieldNameMap = this.createFormFieldsToPersistListing(
+            fieldsToInclude,
+            fieldsToExclude
+        );
+
+        PersistenceProvider.create({
+            persistOptions: {
+                path,
+                ...rootPersistWith
+            },
+            target: {
+                getPersistableState: () => {
+                    return new PersistableState(
+                        fieldNameMap.reduce(
+                            (acc, name) => ({
+                                ...acc,
+                                [name]: this.serialize(this.fields[name].value)
+                            }),
+                            {}
+                        )
+                    );
+                },
+                setPersistableState: ({value: formValues}) => {
+                    // There is no metadata on a field to denote it is a date.
+                    // Use a regex matcher to tests for dates and format matches accurately.
+                    this.setValues(
+                        fieldNameMap.reduce((acc, name) => {
+                            return {...acc, [name]: this.deserialize(formValues[name])};
+                        }, {} as PlainObject)
+                    );
+                }
+            },
+            owner: this
+        });
+    }
+
+    private serialize(formValue: unknown) {
+        if (isArray(formValue)) return formValue.map(this.serialize);
+        if (isDate(formValue)) return {__type: 'date', value: formValue.toJSON()};
+        if (isLocalDate(formValue)) return {__type: 'localDate', value: formValue.toJSON()};
+        if (isObject(formValue)) return mapValues(formValue, this.serialize);
+        return formValue;
+    }
+
+    private deserialize(formValue: unknown) {
+        if (isArray(formValue)) return formValue.map(this.deserialize);
```

Comment:
Same question as above re passing a reference to the method itself when not using an arrow function

Reply:
Fixed